src/lib/sundials_includes.pxd

```diff
@@ -369,13 +416,19 @@ IF SUNDIALS_VERSION >= (3,0,0):
     cdef extern from "cvodes/cvodes_direct.h":
         ctypedef int (*CVDlsDenseJacFn)(realtype t, N_Vector y, N_Vector fy, 
                        SUNMatrix Jac, void *user_data, N_Vector tmp1, N_Vector tmp2, N_Vector tmp3)
-        int CVDlsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS, SUNMatrix A)
-        int CVDlsSetJacFn(void *cvode_mem, CVDlsDenseJacFn djac)
+        IF SUNDIALS_VERSION >= (4,0,0):
+            int CVodeSetLinearSolver(void *cvode_mem, SUNLinearSolver LS, SUNMatrix A)
+            int CVodeSetJacFn(void *cvode_mem, CVDlsDenseJacFn djac)
+        ELSE:
+            int CVDlsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS, SUNMatrix A)
+            int CVDlsSetJacFn(void *cvode_mem, CVDlsDenseJacFn djac)
     cdef extern from "cvodes/cvodes_spils.h":
-        int CVSpilsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS)
         ctypedef int (*CVSpilsJacTimesSetupFn)(realtype t, N_Vector y, N_Vector fy, void *user_data)
-        int CVSpilsSetJacTimes(void *cvode_mem, CVSpilsJacTimesSetupFn jtsetup, CVSpilsJacTimesVecFn jtimes)
-
+        IF SUNDIALS_VERSION >= (4,0,0):
+            int CVodeSetJacTimes(void *cvode_mem, CVSpilsJacTimesSetupFn jtsetup, CVSpilsJacTimesVecFn jtimes)
```

Comment:
Should there be an extra line 

int CVSpilsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS)

or is this no longer required here?

Reply:
It is only used for sundials=3.x